codegen/src/generator/ec2.rs

```diff
@@ -417,7 +417,7 @@ fn generate_struct_field_serializers(shape: &Shape) -> String {
                 ",
                 field_name = generate_field_name(member_name),
                 member_shape_name = member.shape,
-                tag_name = member.tag_name(),
+                tag_name = member.location_name.clone().unwrap_or_else(|| member_name.clone()),
```

Comment:
The exact same piece of logic can be found on line 293, and while we're at it should also be used just above here on line 406 in the `if` branch of this `if/else` block (currently still just `tag_name = member_name`).

Since the same piece of logic applies in at least 3 places in this file alone, we should pull it up into a single method and use that instead of duplicating the code.  The ideal place would be in this `impl Member{}` block - https://github.com/rusoto/rusoto/blob/master/codegen/src/botocore.in.rs#L167